tests/workload/loader_test.py

```diff
@@ -289,6 +289,87 @@ def test_download_document_archive_if_no_file_available(self, is_file, get_size,
                                     "/tmp/docs.json.bz2", 200, progress_indicator=mock.ANY)
         prepare_file_offset_table.assert_called_with("/tmp/docs.json")
 
+    @mock.patch("osbenchmark.utils.io.prepare_file_offset_table")
+    @mock.patch("osbenchmark.utils.io.decompress")
+    @mock.patch("osbenchmark.utils.net.download")
+    @mock.patch("osbenchmark.utils.io.ensure_dir")
+    @mock.patch("os.path.getsize")
+    @mock.patch("os.path.isfile")
+    def test_download_document_archive_with_source_url_compressed(self, is_file, get_size, ensure_dir, download, decompress,
+                                                                  prepare_file_offset_table):
+        # uncompressed file does not exist
+        # compressed file does not exist
+        # after download compressed file exists
+        # after download uncompressed file still does not exist (in main loop)
+        # after download compressed file exists (in main loop)
+        # after decompression, uncompressed file exists
+        is_file.side_effect = [False, False, True, False, True, True, True]
+        # compressed file size is 200 after download
+        # compressed file size is 200 after download (in main loop)
+        # uncompressed file size is 2000 after decompression
+        # uncompressed file size is 2000 after decompression (in main loop)
+        get_size.side_effect = [200, 200, 2000, 2000]
+
+        prepare_file_offset_table.return_value = 5
+
+        p = loader.DocumentSetPreparator(workload_name="unit-test",
+                                         downloader=loader.Downloader(offline=False, test_mode=False),
+                                         decompressor=loader.Decompressor())
+
+        p.prepare_document_set(document_set=workload.Documents(source_format=workload.Documents.SOURCE_FORMAT_BULK,
+                                                            base_url="http://benchmarks.elasticsearch.org/corpora",
+                                                    source_url="http://benchmarks.elasticsearch.org/corpora/unit-test/docs.json.bz2",
```

Comment:
Nit: Although this is for a unittest, is there another link we could use to test it out with instead of one from elastic?

Reply:
That's a good point.  These were cloned from other tests that have that link.  Will go through all and replace them in a separate commit.